File: Backend/demo/src/main/java/com/example/paintapp/shapes/Triangle.java
package com.example.paintapp.shapes;

import com.example.paintapp.shapes.base.Shape;

public class Triangle extends Shape {
    public Triangle() {
        this.type = "Triangle";
    }

    //For Debugging
    @Override
    public void draw() {
        System.out.println("Drawing Triangle at (" + x1 + "," + y1 + ") with properties: " + properties);
    }
}

File: Frontend/src/app/models/triangle.ts
import { Shape } from './shape';

export class Triangle extends Shape {
  constructor(
    x1: number,
    y1: number,
    x2: number,
    y2: number,
    properties: { [key: string]: any } = {}
  ) {
    super('triangle', x1, y1, x2, y2, properties);
  }
}
